### PR TITLE
Dashboard Cards: Pages card skeleton

### LINKED ISSUE #20439: Dashboard Cards: Add Pages card in correct position
## Tasks

- [ ] Add a skeleton card to the dashboard
- [ ] Implement hiding logic if Pages is not supported
- [ ] Add card in the correct position

### PR DESCRIPTION
**Pull request by hassaanelgarem**
Depends on #20477
Closes #20439 
Part of #20438 

## Description

Adds a skeleton Pages card to the dashboard. The PR also adds some logic and tests related to handling changes to the dashboard endpoint.

## Testing Instructions

1. Run the Jetpack app
2. Enable the Pages Card Remote Feature Flag from the debug Menu
3. Enable the Personalize Home Tab Feature Flag from the debug Menu
4. Navigate to the dashboard
5. Make sure that the Pages card is added in the correct position
6. Tap on "Personalize your home tab"
7. Make sure the Pages card shows up
8. Make sure that hiding and reshowing the new card works as expected

## Regression Notes
1. Potential unintended areas of impact
N/A

2. What I did to test those areas of impact (or what existing automated tests I relied on)
N/A

3. What automated tests I added (or what prevented me from doing so)
Added tests to `BlogDashboardServiceTests` and `DashboardCardTests`

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/Classes/Models/WPAccount.m [hunk 1]
```diff
@@ -1,9 +1,6 @@
 #import "WPAccount.h"
 #import "WordPress-Swift.h"
 
-static NSString * const WordPressComOAuthKeychainServiceName = @"public-api.wordpress.com";
-static NSString * const JetpackComOAuthKeychainServiceName = @"jetpack.public-api.wordpress.com";
-
```
Comment (hassaanelgarem) [inline WordPress/Classes/Models/WPAccount.m:6]: This change is unrelated to this PR. These are unused constants that should've been removed in https://github.com/wordpress-mobile/WordPress-iOS/pull/20472.

File: WordPress/Classes/ViewRelated/Blog/Blog Dashboard/DashboardCard.swift [hunk 2]
```diff
@@ -143,6 +151,8 @@ enum DashboardCard: String, CaseIterable {
     enum RemoteDashboardCard: String, CaseIterable {
         case todaysStats = "todays_stats"
         case posts
+        case pages
+        case activity
```
Comment (hassaanelgarem) [inline WordPress/Classes/ViewRelated/Blog/Blog Dashboard/DashboardCard.swift:155]: Heads up: This change causes the dashboard cards request to fail. It will keep failing until the endpoint is updated to handle the new cards.

File: WordPress/Classes/ViewRelated/Blog/BlogPersonalization/BlogDashboardPersonalizationViewModel.swift [hunk 3]
```diff
@@ -46,6 +46,8 @@ private extension DashboardCard {
             return NSLocalizedString("personalizeHome.dashboardCard.scheduledPosts", value: "Scheduled posts", comment: "Card title for the pesonalization menu")
         case .activityLog:
             return NSLocalizedString("personalizeHome.dashboardCard.activityLog", value: "Recent activity", comment: "Card title for the pesonalization menu")
+        case .pages:
+            return NSLocalizedString("pages.dashboardCard.activityLog", value: "Pages", comment: "Card title for the pesonalization menu")
```
Comment (momo-ozawa) [inline WordPress/Classes/ViewRelated/Blog/BlogPersonalization/BlogDashboardPersonalizationViewModel.swift]:
```suggestion
            return NSLocalizedString("personalizeHome.dashboardCard.pages", value: "Pages", comment: "Card title for the pesonalization menu")
```

File: WordPress/WordPressTest/Dashboard/BlogDashboardServiceTests.swift [hunk 4]
```diff
@@ -86,6 +90,46 @@ class BlogDashboardServiceTests: CoreDataTestCase {
...
+        let blog = newTestBlog(id: wpComID, context: mainContext)
+
+        service.fetch(blog: blog) { cards in
+            let activityCardItem = cards.first(where: {$0.cardType == .activityLog})
+
+            // Activity section exists
+            XCTAssertNotNil(activityCardItem)
+
+            // 2 activity items
+            XCTAssertEqual(activityCardItem!.apiResponse!.pages!.count, 2)
```
Comment (momo-ozawa) [inline WordPress/WordPressTest/Dashboard/BlogDashboardServiceTests.swift]:
```suggestion
            XCTAssertEqual(activityCardItem!.apiResponse!.activity!.count, 2)
```

**Review by momo-ozawa — APPROVED** [review]
Works as described, but found a few typos. I tested by merging `trunk` into my local copy of this branch to get the changes from https://github.com/wordpress-mobile/WordPress-iOS/pull/20472
